pandas/core/groupby.py

```diff
@@ -2315,6 +2315,11 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
```

Comment:
Yes, but in case of `sort=False`, [some expect](https://github.com/pandas-dev/pandas/issues/8868) the result categories sorted in the order of encounter.